backend/src/main/java/de/wehnerts/backend/security/filter/JWTAuthFilter.java

```diff
@@ -0,0 +1,56 @@
+package de.wehnerts.backend.security.filter;
+
+import de.wehnerts.backend.security.service.JWTUtilService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@Component
+public class JWTAuthFilter extends OncePerRequestFilter{
+    private final JWTUtilService jwtUtilService;
+
+    @Autowired
+    public JWTAuthFilter(JWTUtilService jwtUtilService) {
+        this.jwtUtilService = jwtUtilService;
+    }
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)throws ServletException, IOException{
+        String token = getAuthToken(request);
+
+        try {
+            if (token != null && !token.isBlank()){
+                String username = jwtUtilService.extractUsername(token);
+                setContext(username);
+            }else{
+                System.out.println("Token is null or blank.");
```

Comment:
```suggestion
                System.out.println("Token is null or blank.");
```
You should here use a logger